test/controllers/RepresentativeDanControllerSpec.scala

```diff
@@ -0,0 +1,129 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.ControllerSpecBase
+import controllers.actions.FakeDataRetrievalAction
+import forms.RepresentativeDanFormProvider
+import mocks.repositories.MockSessionRepository
+import models.UserAnswers
+import pages.{DefermentAccountPage, DefermentPage, DefermentTypePage, SplitPaymentPage}
+import play.api.http.Status
+import play.api.mvc.{Call, Result}
+import play.api.test.Helpers._
+import views.html.RepresentativeDanView
+
+import scala.concurrent.Future
+
+class RepresentativeDanControllerSpec extends ControllerSpecBase {
+
+  def buildForm(accountNumber: Option[String] = Some("1234567"),
+                danType: Option[String] = Some("A")): Seq[(String, String)] =
+    (
+      (accountNumber.map(_ => "accountNumber" -> accountNumber.get) ++
+        danType.map(_ => "value" -> danType.get)).toSeq
+      )
+
+  trait Test extends MockSessionRepository {
+    private lazy val representativeDanView: RepresentativeDanView = app.injector.instanceOf[RepresentativeDanView]
+
+    val userAnswers: Option[UserAnswers] = Some(UserAnswers("credId"))
+    private lazy val dataRetrievalAction = new FakeDataRetrievalAction(userAnswers)
+
+    val formProvider: RepresentativeDanFormProvider = injector.instanceOf[RepresentativeDanFormProvider]
+    val form: RepresentativeDanFormProvider = formProvider
+
+    MockedSessionRepository.set(Future.successful(true))
+
+    lazy val controller = new RepresentativeDanController(authenticatedAction, dataRetrievalAction, dataRequiredAction,
+      mockSessionRepository, messagesControllerComponents, representativeDanView, form)
+  }
+
+  "GET Representative Dan page" should {
+    "return OK" in new Test {
+      val result: Future[Result] = controller.onLoad(fakeRequest)
+      status(result) mustBe Status.OK
+    }
+
+    "return HTML" in new Test {
+      override val userAnswers: Option[UserAnswers] =
+        Some(
+          UserAnswers("some-cred-id")
+            .set(DefermentTypePage, "A").success.value
+            .set(DefermentAccountPage, "1234567").success.value
+        )
+      val result: Future[Result] = controller.onLoad(fakeRequest)
+      contentType(result) mustBe Some("text/html")
+      charset(result) mustBe Some("utf-8")
+    }
+
+    "the backLink functionality is called" should {
+      "redirect to the deferment page" in new Test {
+        override val userAnswers: Option[UserAnswers] = Some(
+          UserAnswers("some-cred-id").set(DefermentPage, false).success.value
+        )
+        controller.backLink(userAnswers.get) mustBe Call("GET", controllers.routes.DefermentController.onLoad().url)
```

Comment:
You don't need to create a Call object because controllers.routes.DefermentController.onLoad() IS a Call object. Applies in a couple of other places too.